src/exchange/providers/retry_with_back_off_decorator.py

```diff
@@ -0,0 +1,51 @@
+from functools import wraps
+import time
+from typing import Any, Callable, Dict, List, Optional
+from httpx import HTTPStatusError, Response
+
+def retry_with_backoff(
```

Comment:
have we considered using `tenacity` package to do this? 
https://tenacity.readthedocs.io/en/latest/

Reply:
+1

Reply:
Maybe this is simple enough that it's not worth the extra dep?

Reply:
Since we've done some testing already with this implementation, I think we just keep it and benefit from the fewer dependencies. We can always revisit this in the future.

Reply:
Hi @salman1993, @zakiali,

Thanks for asking this question here!  When I working on this PR I was also thinking whether we should use any existing library instead of creating our own.  I also found `tenacity`  at that time and had a look.

Below are the reason that  I did not use the package:
- provides more features and these features are not all required for our use case at the moment.  It might be a bit  overkill to implement the requirements by adding a new dependency
- The package is popular but I was not sure about whether there is any license concern if we used in our open source project.   So it is better to check with the team about the license afterwards

However, if we more complicated requirements on the retry in the future, I think it is still worthwhile considering `tenacity`